tests/Simplify/ListTest.elm

```diff
@@ -4233,6 +4233,159 @@ a = Array.toList >> List.foldl f x
+        , test "should replace List.foldl f init (Dict.values dict) by Dict.foldl (always f) init dict" <|
+            \() ->
+                """module A exposing (..)
+a = List.foldl f init (Dict.values dict)
+"""
+                    |> Review.Test.run ruleWithDefaults
+                    |> Review.Test.expectErrors
+                        [ Review.Test.error
+                            { message = "To fold over dict values, you don't need to convert to a list"
+                            , details =
+                                [ "You can replace these calls by Dict.foldl and ignore the first incoming value in the reduce function." ]
+                            , under = "List.foldl"
+                            }
+                            |> Review.Test.whenFixed """module A exposing (..)
+a = Dict.foldl (always f) init dict
+"""
+                        ]
+        , test "should replace List.foldl (\\a s -> f a s) init << Dict.values by Dict.foldl (\\_ a s -> f a s) init" <|
+            \() ->
+                """module A exposing (..)
+a = List.foldl (\\a s -> f a s) init << Dict.values
+"""
+                    |> Review.Test.run ruleWithDefaults
+                    |> Review.Test.expectErrors
+                        [ Review.Test.error
+                            { message = "To fold over dict values, you don't need to convert to a list"
+                            , details =
+                                [ "You can replace this composition by Dict.foldl and ignore the first incoming value in the reduce function." ]
+                            , under = "List.foldl"
+                            }
+                            |> Review.Test.whenFixed """module A exposing (..)
+a = Dict.foldl (\\_ a s -> f a s) init
+"""
+                        ]
+        , test "should replace Dict.values >> List.foldl (\\a s -> f a s) init by Dict.foldl (\\_ a s -> f a s) init" <|
+            \() ->
+                """module A exposing (..)
+a = Dict.values >> List.foldl (\\a s -> f a s) init
+"""
+                    |> Review.Test.run ruleWithDefaults
+                    |> Review.Test.expectErrors
+                        [ Review.Test.error
+                            { message = "To fold over dict values, you don't need to convert to a list"
+                            , details =
+                                [ "You can replace this composition by Dict.foldl and ignore the first incoming value in the reduce function." ]
+                            , under = "List.foldl"
+                            }
+                            |> Review.Test.whenFixed """module A exposing (..)
+a = Dict.foldl (\\_ a s -> f a s) init
+"""
+                        ]
+        , test "should replace List.foldl (\\a s -> f a s) init (Dict.keys dict) by Dict.foldl (\\a _ s -> f a s) init dict" <|
+            \() ->
+                """module A exposing (..)
+a = List.foldl (\\a s -> f a s) init (Dict.keys dict)
```

Comment:
Should we support/have tests for when the there are only some of the lambda parameters, like `\a -> f a`?